### PR TITLE
Request preview with the same size as in the file listing

### PR DESCRIPTION
**Pull request by CarlSchwan**
Potentially will allow bigger caching potential with a few other changes
in the server repo

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Comment by CarlSchwan** [comment]
Failing tests are caused by unrelated change in the server repo with CachedMountFileInfo

File: lib/Controller/APIv2Controller.php [hunk 1]
```diff
@@ -373,14 +373,16 @@ protected function getPreview(string $owner, int $fileId, string $filePath): arr
...
-					'x' => 150,
-					'y' => 150,
+				$preview['source'] = $this->urlGenerator->linkToRouteAbsolute('core.Preview.getPreviewByFileId', [
+					'fileId' => $fileId,
+					'c' => $fileInfo->getEtag(),
+					'x' => 250,
+					'y' => 250,
+					'forceIcon' => 0,
+					'a' => 0,
 				]);
```
Comment (nickvergessen) [inline lib/Controller/APIv2Controller.php:386]: https://nextcloud24.local/index.php/core/preview?fileId=690&c=de6528a27e95f41c3ada4606f337b983&x=250&y=250&forceIcon=0&a=0
https://nextcloud24.local/index.php/core/preview?c=de6528a27e95f41c3ada4606f337b983&forceIcon=0&a=0&x=250&y=250&fileId=690

URLs still don't match for me so browser cache will not hit?

**Comment by nickvergessen** [comment]
Failures are new in this branch:
```
There were 4 failures:

1) OCA\Activity\Tests\Controller\APIv2ControllerTest::testGetPreview with data set #0 ('author', 42, '/path', '/currentPath', true, true, false, '/preview/dir', true, 'dir')
Failed asserting that false is identical to Array &0 (
    'link' => '/preview/currentPath'
    'source' => '/preview/dir'
    'mimeType' => 'dir'
    'isMimeTypeIcon' => true
    'fileId' => 42
    'view' => 'files'
    'filename' => 'path'
).

/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/Controller/APIv2ControllerTest.php:772

2) OCA\Activity\Tests\Controller\APIv2ControllerTest::testGetPreview with data set #1 ('author', 42, '/file.txt', '/currentFile.txt', false, true, false, '/preview/mpeg', true, 'audio/mp3')
Failed asserting that false is identical to Array &0 (
    'link' => '/preview/currentFile.txt'
    'source' => '/preview/mpeg'
    'mimeType' => 'audio/mp3'
    'isMimeTypeIcon' => true
    'fileId' => 42
    'view' => 'files'
    'filename' => 'file.txt'
).

/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/Controller/APIv2ControllerTest.php:772

3) OCA\Activity\Tests\Controller\APIv2ControllerTest::testGetPreview with data set #2 ('author', 42, '/file.txt', '/currentFile.txt', false, true, true, '/preview/currentFile.txt', false, 'text/plain')
Failed asserting that false is identical to Array &0 (
    'link' => '/preview/currentFile.txt'
    'source' => '/preview/currentFile.txt'
    'mimeType' => 'text/plain'
    'isMimeTypeIcon' => false
    'fileId' => 42
    'view' => 'files'
    'filename' => 'file.txt'
).

/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/Controller/APIv2ControllerTest.php:772

4) OCA\Activity\Tests\Controller\APIv2ControllerTest::testGetPreview with data set #3 ('author', 42, '/file.txt', '/currentFile.txt', false, false, true, 'source::getPreviewFromPath', true, 'text/plain')
Failed asserting that false is identical to Array &0 (
    'link' => '/preview/currentFile.txt'
    'source' => 'source::getPreviewFromPath'
    'mimeType' => 'text/plain'
    'isMimeTypeIcon' => true
    'fileId' => 42
    'view' => 'files'
    'filename' => 'file.txt'
).

/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/Controller/APIv2ControllerTest.php:772

ERRORS!
```

The errors also happen in master: #745 
```
There were 2 errors:

1) OCA\Activity\FilesHooksTest::testAddNotificationsForFileAction with data set #2 (array(array(42), array(false)), true, true, array(array('restored_self', array(array('/path')), '/path', false, 42), array('restored_by', array(array('/path'), 'user'), '/path', false, 0)))
ArgumentCountError: Too few arguments to function OC\Files\Config\CachedMountFileInfo::__construct(), 7 passed in /home/nickv/Nextcloud/24/server/appspackaged/activity/tests/FilesHooksTest.php on line 421 and exactly 8 expected

/home/nickv/Nextcloud/24/server/lib/private/Files/Config/CachedMountFileInfo.php:33
/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/FilesHooksTest.php:421

2) OCA\Activity\FilesHooksTest::testAddNotificationsForFileAction with data set #3 (array(array(), array(true)), true, true, array(array('restored_self', array(array('/path')), '/path', false, 0), array('restored_by', array(array('/path'), 'user'), '/path', true, 0)))
ArgumentCountError: Too few arguments to function OC\Files\Config\CachedMountFileInfo::__construct(), 7 passed in /home/nickv/Nextcloud/24/server/appspackaged/activity/tests/FilesHooksTest.php on line 421 and exactly 8 expected

/home/nickv/Nextcloud/24/server/lib/private/Files/Config/CachedMountFileInfo.php:33
/home/nickv/Nextcloud/24/server/appspackaged/activity/tests/FilesHooksTest.php:421
```

File: lib/Controller/APIv2Controller.php (hunk 1 above)
Comment (CarlSchwan) [reply lib/Controller/APIv2Controller.php:386]: Interesting the order is not the same as I'm observing locally:

activity: https://nextcloud.local/index.php/core/preview?forceIcon=0&a=0&x=250&y=250&fileId=5382&c=6217b04be4cce

file list (without https://github.com/nextcloud/server/pull/31346): /index.php/core/preview?fileId=5351&c=6217b04986bda&x=250&y=250&forceIcon=0&a=0

file list (with https://github.com/nextcloud/server/pull/31346): https://nextcloud.local/index.php/core/preview?forceIcon=0&a=0&x=250&y=250&fileId=5351&c=6217b04986bda

I wonder if it depends on the PHP version. I use PHP 7.4.

Comment (CarlSchwan) [reply lib/Controller/APIv2Controller.php]: Ok I found out the reasons, we were doing a asort($parameters) in server + some caching in redis. It should be good now with the latest version of https://github.com/nextcloud/server/pull/31346

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**